src/systemctl/systemctl-list-unit-files.c

```diff
@@ -176,7 +178,7 @@ int verb_list_unit_files(int argc, char *argv[], void *userdata) {
 
                 UnitFileList *u;
                 HASHMAP_FOREACH(u, h) {
-                        if (!output_show_unit_file(u, NULL, NULL))
+                        if (output_show_unit_file(u, NULL, NULL) <= 0)
```

Comment:
i don't think we should continue here if the call fails with an error. Just do 

```c
r = output_show_unit_file(u, /* states= */ NULL, /* patterns= */ NULL);
if (r < 0)
        return log_error_errno(r, "Failed to determine whether to output unit file entry: %m");
        if (!r)
                continue;
```

or so.